src/meltano/core/settings_service.py

```diff
@@ -374,9 +374,9 @@ def get_with_metadata(  # noqa: WPS210, WPS615
                 value = expanded_value
 
         if setting_def:
-            if (
-                setting_def.kind == SettingKind.OBJECT
-                and metadata["source"] is SettingValueStore.DEFAULT
+            if setting_def.kind == SettingKind.OBJECT and (
+                metadata["source"]
+                in {SettingValueStore.DEFAULT, SettingValueStore.INHERITED}
```

Comment:
I can't believe how many hours I put in for a fix that seems too simple 😳

Reply:
> Would there potentially be other places where nested (or "object") extras are being ignored by this? For instance, should this also include `SettingValueStore.DB`? I can't immediately think of a case where that could happen but this logic isn't totally clear to me.

@cjohnhanson AFAICT, other stores like env vars and database are safe because the value comes in a string and cast into an object by `SettingDefinition.cast_value`. All of these extras of kind `object` have a default value of `{}`, and the original check was added to allow things like `_metadata.stream_id.replication-key` to be expanded after trying to find values for `_metadata` in all other sources before `DEFAULT`.

A comment is a good idea 👍